File: springboot-backend/src/main/java/com/capstone/project/service/AssignmentService.java
package com.capstone.project.service;

import com.capstone.project.exception.ResourceNotFroundException;
import com.capstone.project.model.Assignment;

import java.text.ParseException;
import java.util.List;
import java.util.Map;

public interface AssignmentService {

    List<Assignment> getAllAssignment();
    List<Assignment> getAllAssignmentByClassId(int id);

    Assignment createAssignment(Assignment assignment, List<String> file, int type);

    Assignment getAssignmentById (int id) throws ResourceNotFroundException;

    Assignment updateAssignment(int id, Assignment assignment, List<String> files, int type) throws ResourceNotFroundException;

    Boolean deleteAssignment(int id) throws ResourceNotFroundException;

    Map<String, Object> getFilterAssignment( String search, String author, String from, String to,int classid, int page, int size) throws ResourceNotFroundException;
}

File: react-frontend/src/services/AssignmentService.js
import axios from 'axios'

const API_BASE_URL = process.env.REACT_APP_HOST

const createAssignment = (assignment) => {
    return axios.post(API_BASE_URL + '/assignments', assignment)
}

const getAssignmentById = (id) => {
    return axios.get(API_BASE_URL + '/assignments/' + id)
}

const updateAssignment = (id, assignment) => {
    return axios.put(API_BASE_URL + '/assignments/' + id, assignment)
}

const deleteAssignment = (id) => {
    return axios.delete(API_BASE_URL + '/assignments/' + id)
}

const getFilterList = (
    search,
    author,
    fromStart,
    toStart,
    fromCreated,
    toCreated,
    duedatefrom,
    duedateto,
    draft,
    direction,
    sortBy,
    classId,
    page,
    size
) => {
    return axios.get(
        API_BASE_URL +
            '/filterassignment?search' +
            search +
            '&author' +
            author +
            (fromStart ? `&fromstarted${fromStart}` : '') +
            (toStart ? `&tostarted${toStart}` : '') +
            (fromCreated ? `&fromcreated${fromCreated}` : '') +
            (toCreated ? `&tocreated${toCreated}` : '') +
            (duedatefrom ? `&duedatefrom${duedatefrom}` : '') +
            (duedateto ? `&duedateto${duedateto}` : '') +
            '&draft' +
            draft +
            '&direction' +
            direction +
            '&sortby' +
            sortBy +
            '&classid' +
            classId +
            '&page' +
            page +
            '&size' +
            size
    )
}

const getNumSubmitAssignment = (assignmentId, classId) => {
    return axios.get(
        API_BASE_URL +
            '/getsubmitassignment?assignmentid=' +
            assignmentId +
            '&classid=' +
            classId
    )
}

const AssignmentService = {
    createAssignment,
    getFilterList,
    getAssignmentById,
    updateAssignment,
    deleteAssignment,
    getNumSubmitAssignment,
}

export default AssignmentService
